pages/_document.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
```

Comment:
Let's add a comment explaining why this file exists.

```suggestion
/*
This _document.tsx file is processed server side only and prevents pages 
from flickering when Material UI is rendered via NextJS. It does that by 
collecting the CSS files generated by Material UI and injecting them into 
the document as a string. 
*/

import * as React from 'react';
```